### PR TITLE
Collections & workbooks sorting fix. Added filters to local storage.

### PR DESCRIPTION
**Pull request by stankis**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
   "name": "@datalens-platform/datalens-ui",
-  "version": "0.0.0",
+  "version": "0.1016.0",
```
Comment (imsitnikov) [inline package-lock.json:3]: Unnecessary changes

File: src/ui/components/CollectionFilters/CollectionFilters.tsx [hunk 2]
```diff
@@ -26,6 +26,10 @@ export enum SortType {
 }
 
 export const collectionPageViewModeStore = 'collectionPageViewMode';
+export const collectionPageFilterOnlyMyStore = 'collectionPageFilterOnlyMyStore';
```
Comment (imsitnikov) [inline src/ui/components/CollectionFilters/CollectionFilters.tsx:29]: Can we store all filters as JSON in one localStorage field – `collectionPageFilters`?

File: src/ui/units/collections/components/CollectionContent/CollectionContent.tsx [hunk 3]
```diff
@@ -235,7 +234,7 @@ export const CollectionContent = React.memo<Props>(
                                     description: item?.description ?? '',
                                     onApply: (collection: UpdateCollectionResponse | null) => {
                                         if (collection) {
-                                            dispatch(updateCollectionInItems(collection));
+                                            refreshContent();
```
Comment (imsitnikov) [inline src/ui/units/collections/components/CollectionContent/CollectionContent.tsx:237]: After renaming a collection or workbook in the list, we make two unnecessary requests:
image

**Review by imsitnikov — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 1 above)
Comment (stankis) [reply package-lock.json]: agree, removed

File: src/ui/components/CollectionFilters/CollectionFilters.tsx (hunk 2 above)
Comment (stankis) [reply src/ui/components/CollectionFilters/CollectionFilters.tsx]: good idea, done

File: src/ui/units/collections/components/CollectionContent/CollectionContent.tsx (hunk 3 above)
Comment (stankis) [reply src/ui/units/collections/components/CollectionContent/CollectionContent.tsx]: agree, fixed

File: src/ui/units/collections/components/CollectionContent/CollectionContent.tsx [hunk 4]
```diff
@@ -343,7 +344,7 @@ export const CollectionContent = React.memo<Props>(
                                     workbookId: item.workbookId,
                                     workbookTitle: item.title,
                                     initialCollectionId: item.collectionId,
-                                    onApply: refreshContent,
+                                    onApply: refreshPage,
```
Comment (imsitnikov) [inline src/ui/units/collections/components/CollectionContent/CollectionContent.tsx:347]: It's not related to the current task, but we can use `onApply: refreshContent` in dialogs `DIALOG_MOVE_COLLECTION` and `DIALOG_MOVE_WORKBOOK`. This will reduce amount of unnecessary requests.

**Review by imsitnikov — CHANGES_REQUESTED** [review]

File: src/ui/utils/filterStorage.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import {GetCollectionContentMode} from 'shared/schema';
+import {OrderBasicField, OrderDirection} from 'shared/schema/us/types/sort';
+import {Utils} from 'ui';
+
+const KEY = 'collectionPageFilters';
+
+type Filter = {
```
[the remark this answers is not in the document]
Comment (stankis) [reply src/ui/utils/filterStorage.ts:7]: renamed

**Review by imsitnikov — APPROVED** [review]
